client-index/src/auto_sync.rs

```diff
@@ -0,0 +1,118 @@
+//!
+//! automatic sync
+//!
+//! how to use
+//! 1. create auto-sync & run
+//! let autosync: AutoSync::new();
+//! autosync.run(url, tendermint_client, storage.clone(), block_handler);
+//!
+//! 2. unlock wallet
+//! autosync.add_wallet(request.name, view_key, private_key, staking_addresses);
+//!
+
+use crate::auto_sync_data::{AddWalletCommand, AutoSyncData, AutoSyncDataShared};
+use crate::auto_synchronizer::AutoSynchronizer;
+use crate::BlockHandler;
+use chain_core::state::account::StakedStateAddress;
+use client_common::tendermint::Client;
+use client_common::Result;
+use client_common::Storage;
+use client_common::{PrivateKey, PublicKey};
+use serde_json::json;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use websocket::OwnedMessage;
+#[derive(Clone, Debug)]
+/// facade for auto sync manager
+pub struct AutoSync {
+    data: AutoSyncDataShared,
+}
+
+impl Default for AutoSync {
```

Comment:
You can use `#[derive(Default)]` instead of implementing this.

Reply:
ok